### PR TITLE
Fixes to the UI and database cleanup

### PR DESCRIPTION
**Pull request by ErikBjare**

### CODE REVIEW

File: src/dashboard/CreatorCard.vue [hunk 1]
```diff
@@ -17,19 +16,20 @@ b-card(class="mb-2" no-body)
...
-        | Edit 
-        font-awesome-icon(icon="edit")
+        | #[font-awesome-icon(icon="edit")] Edit
 
       table.table.table-sm.table-hover.mt-3.mb-0(v-if="showDetails")
         tr
           th Page
           th Duration
+        tr(v-if="activities.length == 0" colspan="0")
+          td No activity found for creator
```
Comment (Powersource) [inline src/dashboard/CreatorCard.vue]: This looks a bit weird (the bar)
image

File: src/lib/db.js [hunk 2]
```diff
@@ -4,35 +4,64 @@ import _ from 'lodash';
...
+  constructor(table, key) {
+    modelAttrs[this.constructor] = { table: table, key: key };
+  }
+
+  async save() {
+    // Does an update if the row already exists, otherwise does a put.
+    let keyname = modelAttrs[this.constructor].key;
+    let key = this[keyname];
+    let table = modelAttrs[this.constructor].table;
+    const exists =
```
Comment (ahnlabb) [inline src/lib/db.js]: Couldn't this just be
``` javascript
const exists = await table.get(this[keyname])
```
which becomes `undefined` when the primary key `this[keyname]` isn't in the store?

Couldn't we also skip all of it and do:
``` js
return table.add(this).catch(err => table.update(key, this))
```
? (also avoids race conditions, however that's not a major concern)

I'm fine either way, just for your consideration.

File: src/lib/db.js [hunk 3]
```diff
@@ -4,35 +4,64 @@ import _ from 'lodash';
...
+      (await table
+        .where(keyname)
+        .equals(this[keyname])
+        .count()) > 0;
+    if (exists) {
+      return table.update(key, this);
+    } else {
+      return this.put();
```
Comment (ahnlabb) [inline src/lib/db.js]: If you're going with this solution couldn't `put` be an `add`?

File: src/lib/db.js (hunk 2 above)
Comment (ErikBjare) [reply src/lib/db.js]: Ah, nice. Fixed.

File: src/lib/db.js (hunk 3 above)
Comment (ErikBjare) [reply src/lib/db.js]: Yeah, but did as you suggested.

**Pull request merged**